default/python/AI/InvasionAI.py

```diff
@@ -543,27 +527,45 @@ def evaluate_invasion_planet(planet_id):
     if clear_path:
         planet_score *= 1.5
     debug(
-        " - planet score: %.2f\n"
-        " - planned troops: %.2f\n"
-        " - projected troop cost: %.1f\n"
-        " - threat factor: %s\n"
-        " - planet detail: %s\n"
-        " - popval: %.1f\n"
-        " - bldval: %s\n"
-        " - enemyval: %s",
-        planet_score,
-        planned_troops,
-        troop_cost,
-        threat_factor,
-        detail,
-        colony_base_value,
-        bld_tally,
-        enemy_val,
+        f" - planet score: {planet_score:.2f}\n"
+        f" - planned troops: {planned_troops:.2f}\n"
+        f" - projected troop cost: {troop_cost:.1f}\n"
+        f" - threat factor: {threat_factor}\n"
+        f" - planet detail: {detail}\n"
+        f" - popval: {colony_base_value:.1f}\n"
+        f" - bldval: {bld_tally}\n"
+        f" - enemyval: {enemy_val}\n"
+        f" - system secured: {system_secured}"
     )
-    debug(" - system secured: %s" % system_secured)
     return [planet_score, planned_troops]
 
 
+def secure_system(system_id: SystemId, secure_now: bool):
```

Comment:
You could specify return type with annotations.
```suggestion
def secure_system(system_id: SystemId, secure_now: bool) -> bool:
```